### DISCUSSION TITLE
How to determine if a collection row exists

### DISCUSSION THREAD

**Discussion by rogerpence** [pocketbase/pocketbase#5055] [Q&A] [answered]
The `getFirstListItem` method throws an uncatchable error (try/catch doesn't catch the error) if the row does not exist. How can you determine if a row exist (based on a column value, not the id)?

```
 const result = await pb.collection("family").getFirstListItem(`name = '${name}'`);
```

**Comment by ganigeorgiev** [comment, marked as answer]
> The getFirstListItem method throws an uncatchable error...

I don't understand what do you mean by this. Wrapping the async call with `try/catch` should work:

```js
try {
     const result = await pb.collection("family").getFirstListItem(`name = '${name}'`);
} catch (err) {
    console.log("Error:", err, err.originalError)
}
```

It it doesn't work, then there is something else going on in your code but without some reproducible steps I'm not sure how to help.

**Reply by ganigeorgiev** [reply]
_See also [JS SDK - Error handling](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#error-handling)._

**Reply by rogerpence** [reply]
thank you very much for answering. I am very embarrassed--I was doing something very stupid. Thanks again.
